### PR TITLE
Skip tags that are not Semantically Versioned

### LINKED ISSUE #75: Handle tags that do not follow semantic versioning
Your documentation indeed points out that tags need to follow SemVer:

```
Note that fetch assumes git tags are specified according to Semantic Versioning principles.
```

However, git tags that are not semantically versioned are valid tags, and if one exists in the source repository, the tool breaks, even if you are not requesting a `--tag`.

I suggest that tags should not be required to be semantically versioned unless a Tag Contraint Expression is specified.

This would allow the tool to work in repositories where non-SemVar tags exist.

To replicate:

```
fetch $ git rev-parse HEAD
83ed55048791ec5385d4143a78798e2a6c82b9f8
```

```
fetch $ ./fetch --repo=https://github.com/pete0emerson/demo --branch=main demo
Downloading latest commit from branch "main" of https://github.com/pete0emerson/demo ...
Extracting files from <repo>/ to demo ...
Download and file extraction complete.
```

```
demo $ git tag banana
demo $ git push --tags
Total 0 (delta 0), reused 0 (delta 0)
To https://github.com/pete0emerson/demo.git
 * [new tag]         banana -> banana
 ```

```
fetch $ ./fetch --repo=https://github.com/pete0emerson/demo --branch=main demo
ERROR: Error occurred while computing latest tag that satisfies version contraint expression: -1 - Malformed version: banana
```

### PR DESCRIPTION
**Pull request by pete0emerson**
This closes #75 

I validated each tag, discarding any that aren't SemVer.

For testing, I suggest that a non-SemVer tag be added to the public and private test repositories `github_test.go`.

https://github.com/gruntwork-io/fetch/blob/4bec67f9ab28423e6b6f72b2fc9d397e5681c671/github_test.go#L25-L28

### CODE REVIEW

File: github.go [hunk 1]
```diff
@@ -126,7 +127,9 @@ func FetchTags(githubRepoUrl string, githubToken string, instance GitHubInstance
 	}
 
 	for _, tag := range tags {
-		tagsString = append(tagsString, tag.Name)
+		if _, err := version.NewVersion(tag.Name); err == nil {
```
Comment (brikis98) [inline github.go:130]: NIT: Add a comment here explaining we want to skip tags that aren't semantically versioned so they don't cause errors. Could even cross-link to the issue you filed.

**Review by brikis98 — COMMENTED** [review]
Thanks for the PR! Could you add an automated test to validate this behavior?

**Comment by pete0emerson** [comment]
> Could you add an automated test to validate this behavior?

I've added `expectedNumTags` and set it to `3` for https://github.com/gruntwork-io/fetch-test-public and `1` for https://github.com/gruntwork-io/fetch-test-private (for which I don't have access, so I can't verify that `1` is correct, but I think it is since `firstReleaseTag` and `lastReleaseTag` are both set to `v0.0.2`).

To fully be confident this is always working, I'd suggest adding a couple of non-SemVer tags to both of those repositories; the expectedNumTags values I've set shouldn't change, but having a "bad" tag in there will then hit the code that skips those tags.

**Comment by brikis98** [comment]
> I've added `expectedNumTags` and set it to `3` for https://github.com/gruntwork-io/fetch-test-public and `1` for https://github.com/gruntwork-io/fetch-test-private (for which I don't have access, so I can't verify that `1` is correct, but I think it is since `firstReleaseTag` and `lastReleaseTag` are both set to `v0.0.2`).
> 
> To fully be confident this is always working, I'd suggest adding a couple of non-SemVer tags to both of those repositories; the expectedNumTags values I've set shouldn't change, but having a "bad" tag in there will then hit the code that skips those tags.

Nice, thanks!

I added a non-semantic tag to both repos that looks like this: https://github.com/gruntwork-io/fetch-test-public/releases/tag/non-semantic

**Review by brikis98 — COMMENTED** [review]
Code LGTM! Could you pull in the latest from `master`, and I'll kick off tests?

**Comment by pete0emerson** [comment]
Done, and my local tests are passing until I hit the private repos.

**Review by brikis98 — APPROVED** [review]
Code LGTM! I'll kick off tests now.

**Comment by brikis98** [comment]
Tests passed! Merging now.

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/fetch/releases/tag/v0.3.11

**Pull request merged**